alpha/lib/model/UserLoginDataPeer.php

```diff
@@ -737,19 +737,53 @@ public static function loginAttemptsLogic($loginData)
 		{
 			throw new kUserException('', kUserException::LOGIN_BLOCKED);
 		}
+
+		$partner = PartnerPeer::retrieveByPK($loginData->getConfigPartnerId());
 		if ($loginData->getLoginAttempts()+1 >= $loginData->getMaxLoginAttempts())
 		{
-			$loginData->setLoginBlockedUntil( time() + ($loginData->getLoginBlockPeriod()) );
-			$loginData->setLoginAttempts(0);
-			$loginData->save();
+			if ($partner && $partner->getLoginFailTimeframe()) //timeframe is set for this partner
+			{
+				if (time() - $loginData->getFirstLoginFailTime() <= $partner->getLoginFailTimeframe()) //timeframe is not over yet
+				{
+					self::setUserBlockedData($loginData);
+					KalturaLog::notice('User login blocked for login data id ['.$loginData->getId().']');
+					throw new kUserException('', kUserException::LOGIN_RETRIES_EXCEEDED);
+				}
+				else
+				{//timeframe is over, we need to reset the login attempts
```

Comment:
We are in a case where we failed to log in but we don't call 		
$loginData->incLoginAttempts();

something in this change seems too cumbersome to me and even less secure than before
IMHO the addition of the login fail time frame should not reset the failed login attempts if the time frame passed but retain a or logic 
i.e did I reach max login attempts in the time frame or reach it regardless. Otherwise, if I have set a max time frame of 1 minute and max attempts to 20 I have 20 chances every minute to guess the password which security-wise is bad.